js/controllers/grid/queries/QueryFormHandler.js

```diff
@@ -0,0 +1,123 @@
+/**
+ * @defgroup js_controllers_grid_queries
+ */
+/**
+ * @file js/controllers/grid/queries/QueryFormHandler.js
+ *
+ * Copyright (c) 2014-2021 Simon Fraser University
+ * Copyright (c) 2000-2021 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class ReadQueryHandler
+ * @ingroup js_controllers_grid_queries
+ *
+ * @brief Handler for a query form modal
+ *
+ */
+(function($) {
+
+	/** @type {Object} */
+	$.pkp.controllers.grid.queries =
+			$.pkp.controllers.grid.queries || {};
+
+
+
+	/**
+	 * @constructor
+	 *
+	 * @extends $.pkp.controllers.form.CancelActionAjaxFormHandler
+	 *
+	 * @param {jQueryObject} $form The query form element
+	 * @param {Object} options non-default Dialog options
+	 *  to be passed into the dialog widget.
+	 *
+	 *  Options are:
+	 *  - all options documented for the CancelActionAjaxFormHandler.
+	 *  - templateUrl: The URL to retrieve templates from.
+	 */
+	$.pkp.controllers.grid.queries.QueryFormHandler =
+			function($form, options) {
+		this.parent($form, options);
+
+		// Set the URL to retrieve templates from.
+		if (options.templateUrl) {
+			this.templateUrl_ = options.templateUrl;
+		}
+
+		this.textSubject_ = null;
+
+		// Attach form elements events.
+		$form.find('#template').change(
+			this.callbackWrapper(this.selectTemplateHandler_));
+	};
+	$.pkp.classes.Helper.inherits($.pkp.controllers.grid.queries.
+			QueryFormHandler, $.pkp.controllers.form.CancelActionAjaxFormHandler);
+
+
+	//
+	// Private properties
+	//
+	/**
+	 * The URL to use to retrieve template bodies
+	 * @private
+	 * @type {string?}
+	 */
+	$.pkp.controllers.grid.queries.
+			QueryFormHandler.prototype.templateUrl_ = null;
+
+
+	//
+	// Private methods
+	//
+	/**
+	 * Respond to an "item selected" call by triggering a published event.
+	 *
+	 * @param {HTMLElement} sourceElement The element that
+	 *  issued the event.
+	 * @param {Event} event The triggering event.
+	 * @private
+	 */
+	$.pkp.controllers.grid.queries.
+			QueryFormHandler.prototype.selectTemplateHandler_ =
+					function(sourceElement, event) {
+		var $form = this.getHtmlElement();
+		var template = $form.find('[name="template"]');
+		$form.find('[name="subject"]').val(template.find('option:selected').text());
+		this.textSubject_
```

Comment:
Is this intended to not have a `;` at the end? I think that syntax is only compatible with some browsers.

Reply:
Hmm, if I remember well, I was going to save the texts (to recover the last typed texts while switching through the templates), but then I changed my mind, as it would require a button to reset...
So, I don't know why the `textSubject_` is still there :D